sympy/physics/mechanics/linearize.py

```diff
@@ -253,35 +262,42 @@ def linearize(self, op_point=None, A_and_B=False, simplify=False):
 
         Parameters
         ==========
-
         op_point : dict or iterable of dicts, optional
             Dictionary or iterable of dictionaries containing the operating
-            point conditions. These will be substituted in to the linearized
-            system before the linearization is complete. Leave blank if you
-            want a completely symbolic form. Note that any reduction in
-            symbols (whether substituted for numbers or expressions with a
-            common parameter) will result in faster runtime.
-
+            point conditions for all or a subset of the generalized
+            coordinates, generalized speeds, and time derivatives of the
+            generalized speeds. These will be substituted into the linearized
+            system before the linearization is complete. Leave set to ``None``
+            if you want the operating point to be an arbitrary set of symbols.
+            Note that any reduction in symbols (whether substituted for numbers
+            or expressions with a common parameter) will result in faster
+            runtime.
         A_and_B : bool, optional
-            If A_and_B=False (default), (M, A, B) is returned for forming
-            [M]*[q, u]^T = [A]*[q_ind, u_ind]^T + [B]r. If A_and_B=True,
-            (A, B) is returned for forming dx = [A]x + [B]r, where
-            x = [q_ind, u_ind]^T.
-
+            If A_and_B=False (default), (M, A, B) is returned and of
+            A_and_B=True, (A, B) is returned. See below.
         simplify : bool, optional
             Determines if returned values are simplified before return.
             For large expressions this may be time consuming. Default is False.
 
-        Potential Issues
-        ================
+        Returns
+        =======
+        M, A, B : Matrices, ``A_and_B=False``
+            Matrices from the implicit form:
+                ``[M]*[q', u']^T = [A]*[q_ind, u_ind]^T + [B]*r``
+        A, B : Matrices, ``A_and_B=True``
+            Matrices from the explicit form:
+                ``[q_ind', u_ind']^T = [A]*[q_ind, u_ind]^T + [B]*r``
+
+        Notes
+        =====
+
+        Note that the process of solving with A_and_B=True is computationally
+        intensive if there are many symbolic parameters. For this reason, it
+        may be more desirable to use the default A_and_B=False, returning M, A,
+        and B. More values may then be substituted in to these matrices later
+        on. The state space form can then be found as A = P.T*M.LUsolve(A), B =
+        P.T*M.LUsolve(B), where P = Linearizer.perm_mat.
```

Comment:
Do we want the code like `A_and_B=True` and `A = P.T * M.LUsolve(A)` to be displayed as code?

Reply:
I didn't edit it. Can also be done in a new PR.